pkg/controllers/hub/membercluster/controller.go

```diff
@@ -61,8 +60,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 	if mc.DeletionTimestamp.IsZero() {
-		klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("member cluster %s is not being deleted",
-			mc.Name)), "The member cluster should have deletionTimeStamp set to a non-zero/non-nil value")
+		klog.V(3).InfoS("The member cluster is not being deleted, ignore it", "memberCluster", mcObjRef)
```

Comment:
Since this is an extended state information I went with 3 
![image](https://github.com/user-attachments/assets/a7eb841b-5993-41e2-ad8c-4e0b3d5d450d)

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md